docs/tutorial.rst

```diff
@@ -136,19 +136,54 @@ variable in Python.  The same is true for deleting them too.
 
 Very nice.
 
-__xonsh_env__
+Environment Lookup with ``${}``
```

Comment:
I think this section, should still follow the section it now comes after.  The reader should know about the environment before they learn about expression based lookup.